src/bootstrap/src/utils/metrics.rs

```diff
@@ -245,6 +251,21 @@ impl BuildMetrics {
     }
 }
 
+fn get_ci_metadata(ci_env: CiEnv) -> Option<CiMetadata> {
+    if ci_env != CiEnv::GitHubActions {
+        return None;
+    }
+    let Some(workflow_run_id) =
+        std::env::var("GITHUB_WORKFLOW_RUN_ID").ok().and_then(|id| id.parse::<u64>().ok())
+    else {
+        return None;
+    };
+    let Ok(repository) = std::env::var("GITHUB_REPOSITORY") else {
+        return None;
+    };
+    Some(CiMetadata { workflow_run_id, repository })
+}
```

Comment:
```suggestion
fn get_ci_metadata(ci_env: CiEnv) -> Option<CiMetadata> {
    (ci_env == CiEnv::GitHubActions).then(|| {
        let workflow_run_id =
            std::env::var("GITHUB_WORKFLOW_RUN_ID").ok().and_then(|id| id.parse::<u64>().ok())?;

        let repository = std::env::var("GITHUB_REPOSITORY").ok()?;
        CiMetadata { workflow_run_id, repository }
    })
}
```

I prefer this style, but it's up to you to choose 👍

Reply:
Maybe I was moulded too much by imperative languages during my programming career, but I have to say that I found it easier to understand with the explicit condition :sweat_smile: Turning the `let-else`s into `?` is nice though, I'll refactor that.